core-common/src/main/java/org/glassfish/jersey/message/internal/ReaderWriter.java

```diff
@@ -242,9 +245,9 @@ private static byte[] readAllBytes(InputStream inputStream) throws IOException {
      * @throws IOException in case of a write failure.
      */
     public static void writeToAsString(String s, OutputStream out, MediaType type) throws IOException {
-        Writer osw = new OutputStreamWriter(out, getCharset(type));
-        osw.write(s, 0, s.length());
-        osw.flush();
+        try (final Writer osw = new OutputStreamWriter(out, getCharset(type))) {
```

Comment:
I do feel this to be a source of issues. While the try with resources closes the OutputStream, the original flush did not close it.

Reply:
Good catch! I will modify this code so it does not close the stream but only flush it (in fact I wonder why you flush it at all, as this should be the *caller's* job, but anyways, it then does what it did before).

Reply:
@jansupol I fixed it in https://github.com/eclipse-ee4j/jersey/pull/5350/commits/85a1e3d9f85e002798032d1e34103e4da19d648f.